iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractForkConnectDisconnectTest.java

```diff
@@ -0,0 +1,247 @@
+        return network;
+    }
+
+    @Test
+    public void fullyClosedTest() {
+        Network network = createNetwork();
+
+        // Useful elements
+        VoltageLevel.NodeBreakerView topo = network.getVoltageLevel("VL1").getNodeBreakerView();
+        Line line1 = network.getLine("L1");
+        Line line2 = network.getLine("L2");
+        Switch disconnector = network.getSwitch("D_L1");
+
+        // L1 and L2 are fully connected
+        assertTrue(topo.getOptionalTerminal(4).isPresent());
+        assertTerminalsStatus(line1.getTerminals(), true);
+        assertTrue(topo.getOptionalTerminal(5).isPresent());
+        assertTerminalsStatus(line2.getTerminals(), true);
+
+        // Disconnect L1
+        line1.disconnect(SwitchPredicates.IS_OPEN.negate());
+
+        // check that L1 is disconnected while L2 is still connected
+        assertTrue(topo.getOptionalTerminal(4).isPresent());
+        assertTerminalsStatus(line1.getTerminals(), false);
+        assertTrue(topo.getOptionalTerminal(5).isPresent());
+        assertTerminalsStatus(line2.getTerminals(), true);
+
+        // D_L1 should be open
+        assertTrue(disconnector.isOpen());
+    }
+
+    @Test
+    public void forkDisconnectedTestWithLines() {
+        Network network = createNetwork();
+
+        // Useful elements
+        VoltageLevel.NodeBreakerView topo = network.getVoltageLevel("VL1").getNodeBreakerView();
+        Line line1 = network.getLine("L1");
+        Line line2 = network.getLine("L2");
+        Switch disconnectorL1 = network.getSwitch("D_L1");
+        Switch disconnectorL2 = network.getSwitch("D_L2");
+        Switch breaker = network.getSwitch("B_L1_L2");
+
+        // In this case, B_L1_L2 is open
+        breaker.setOpen(true);
+
+        // L1 and L2 are fully connected
+        assertTrue(topo.getOptionalTerminal(4).isPresent());
+        assertTerminalsStatus(line1.getTerminals(), true);
+        assertTrue(topo.getOptionalTerminal(5).isPresent());
+        assertTerminalsStatus(line2.getTerminals(), true);
+
+        // Disconnect L1
+        line1.disconnect(SwitchPredicates.IS_OPEN.negate());
+
+        // check that L1 and L2 are disconnected
+        assertTrue(topo.getOptionalTerminal(4).isPresent());
+        assertTerminalsStatus(line1.getTerminals(), false);
+        assertTrue(topo.getOptionalTerminal(5).isPresent());
+        assertTerminalsStatus(List.of(line2.getTerminal1()), false);
+        assertTerminalsStatus(List.of(line2.getTerminal2()), true);
+
+        // D_L1 should be open but not D_L2
+        assertTrue(disconnectorL1.isOpen());
+        assertFalse(disconnectorL2.isOpen());
+    }
+
+    private void assertTerminalsStatus(List<? extends Terminal> terminals, boolean terminalsShouldBeConnected) {
+        if (terminalsShouldBeConnected) {
+            terminals.forEach(terminal -> assertNotNull(terminal.getBusView().getBus()));
+            terminals.forEach(terminal -> assertTrue(terminal.isConnected()));
```

Comment:
these two lines are equivalent in node breaker topology (at least in the known implementations)

Reply:
yes but I preferred to make it obvious and visible since it's part of the issue